tests/test_lib_tokens_push.py

```diff
@@ -1126,25 +1128,38 @@ def test_15_poll_endpoint(self):
 
         # now check that we receive the challenge when polling
         # since we mock the time we can use the same request data
-        with mock.patch('privacyidea.models.datetime') as mock_dt1,\
-                mock.patch('privacyidea.lib.tokens.pushtoken.datetime') as mock_dt2:
-            mock_dt1.utcnow.return_value = timestamp.replace(tzinfo=None) + timedelta(seconds=15)
-            mock_dt2.now.return_value = timestamp + timedelta(seconds=15)
-            res = PushTokenClass.api_endpoint(req, g)
-        self.assertTrue(res[1]['result']['status'], res)
-        chall = res[1]['result']['value'][0]
-        self.assertEqual(chall['nonce'], challenge, chall)
-        self.assertIn('signature', chall, chall)
-        # check that the signature matches
-        sign_string = u"{nonce}|{url}|{serial}|{question}|{title}|{sslverify}".format(**chall)
-        parsed_stripped_server_pubkey = serialization.load_pem_public_key(
-            to_bytes(self.server_public_key_pem),
-            default_backend())
-        parsed_stripped_server_pubkey.verify(b32decode(chall['signature']),
-                                             sign_string.encode('utf8'),
-                                             padding.PKCS1v15(),
-                                             hashes.SHA256())
-        self.assertFalse(db_challenge.get_otp_status()[1], str(db_challenge))
+        # for this request we use the full app context to trigger
```

Comment:
I would rather put this test in `test_api_ttype.py` since this is basically a change in the API layer.